src/Raven.Client/Server/DatabaseTopology.cs

```diff
@@ -2,27 +2,103 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Raven.Client.Documents.Replication;
 using Sparrow.Json.Parsing;
 
 namespace Raven.Client.Documents
 {
+    public class ConflictSolver
+    {
+        public string DatabaseResovlerId;
+        public Dictionary<string, ScriptResolver> ResolveByCollection;
+        public bool ResolveToLatest;
+
+        public bool ConflictResolutionChanged(ConflictSolver other)
+        {
+            if ((ResolveByCollection == null ^ other.ResolveByCollection == null) == false)
+                return true;
+
+            return ResolveToLatest == other.ResolveToLatest &&
+                   (ResolveByCollection?.SequenceEqual(other.ResolveByCollection) ?? true);
+        }
+
+        public bool IsEmpty()
+        {
+            return ResolveToLatest == false && DatabaseResovlerId == null && ResolveByCollection?.Count == 0;
+        }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(DatabaseResovlerId)] = DatabaseResovlerId,
+                [nameof(ResolveToLatest)] = ResolveToLatest,
+                [nameof(ResolveByCollection)] = new DynamicJsonArray
+                {
+                    ResolveByCollection.Select( item => new DynamicJsonValue
+                    {
+                        [nameof(item.Key)] = item.Value.ToJson()
+                    })
+                }
+            };
+        }
+    }
+
+    public interface IDatabaseTask
+    {        
+    }
+
+    public class DatabaseWatcher : ReplicationNode, IDatabaseTask
+    {
+    }
+
+    public class DatabasePromotable : ReplicationNode, IDatabaseTask
+    {
+    }
+
     public class DatabaseTopology
     {
         public List<string> Members = new List<string>();
         public List<string> Promotables = new List<string>();
-        public List<string> Watchers = new List<string>();
+        public List<DatabaseWatcher> Watchers = new List<DatabaseWatcher>();
+
+        public Dictionary<string,string> NameToUrlMap = new Dictionary<string, string>();
+
+        // If we want to prevent from a replication we set an entry 
+        // e.g. setting ("A","C") will prevent replication from server A to C for this database.
+        public Dictionary<string,string> CustomConnectionBlocker = new Dictionary<string, string>();
```

Comment:
What is the usage scenario for that? Because the way to handle _that_ is to remove the node from the db entirely. 
We _want_ this to be simpler, by default master/master only. With watchers as the escape hatch